net/respondd-module-airtime/src/respondd.c

```diff
@@ -0,0 +1,58 @@
+#include <string.h>
+#include <stdio.h>
+#include <json-c/json.h>
+#include <respondd.h>
+
+#include "airtime.h"
+#include "ifaces.h"
+
+void fill_airtime_json(struct airtime_result *air, struct json_object *wireless) {
+	struct json_object *obj = NULL;
+
+	obj = json_object_new_object();
+	if(!obj)
+		return;
+
+	json_object_object_add(obj, "frequency", json_object_new_int(air->frequency));
+	json_object_object_add(obj, "active",    json_object_new_int64(air->active_time));
+	json_object_object_add(obj, "busy",      json_object_new_int64(air->busy_time));
+	json_object_object_add(obj, "rx",        json_object_new_int64(air->rx_time));
+	json_object_object_add(obj, "tx",        json_object_new_int64(air->tx_time));
+	json_object_object_add(obj, "noise",     json_object_new_int(air->noise));
+
+	json_object_array_add(wireless, obj);
+}
+
+static struct json_object *respondd_provider_statistics(void) {
+	struct airtime_result airtime = {};
+	struct json_object *result, *wireless;
+	struct iface_list *ifaces;
+	void *freeptr;
```

Comment:
Please move the definition of *freeptr* into the while loop below to make its scope clearer.